src/Pages/HomePage/HomePage.js

```diff
@@ -1,15 +1,52 @@
-import React from 'react'
-import { Link, Outlet } from 'react-router-dom'
+import React, { useEffect, useState } from 'react'
+import { VideoCard } from '../../Components/VideoCard/VideoCard'
+import { CategoryChip } from './HomeComponent/CategoryChip'
+import { fetchVideos } from '../../Utils/fetch-videolist'
+import './homepage.css'
+import { useCategory } from '../../Context/category-provider'
+
 
 const HomePage = () => {
+    const [videoList, setvideoList] = useState([]);
+    const [whiteBg, setWhiteBg] = useState(false)
+    const { videoCategory } = useCategory()
+
+    useEffect(() => {
+        (async () => {
+            const { data, errorData } = await fetchVideos();
+            !errorData[0] ? setvideoList(data) : console.log(errorData[1])
+        })()
+    }, []);
+
+    const categoryList = [
+        { title: 'All', lableFor: 'all' }, { title: 'New Updates', lableFor: 'updates' }, { title: 'Montages', lableFor: 'montages' }, { title: 'Funny Clips', lableFor: 'funny' }, { title: 'Tournaments', lableFor: 'tournaments' },]
+
     return (
         <div>
-            <h1>Template Routing Common ui</h1>
-            <Link to="/test1">test1</Link>
-            <Link to="/test2">test2</Link>
-            <Link to="/test3">test3</Link>
-            <Outlet />
+            <div className="hero-section center-items">
+                <h1 className='center-txt'>THE <span className='violet-txt'>VIDEO GAMING </span> ARCHIVE</h1>
+            </div>
+            <div className="center-items">
+                <div className={whiteBg ? "nav-search-bar hide-sm mg-t-20 bg-white" : "nav-search-bar hide-sm mg-t-20"}>
+                    <input onFocus={() => setWhiteBg(true)} onBlur={() => setWhiteBg(false)} className="search-bar" type="text" placeholder="Search For Videos..." />
+                    <i className="fa fa-brands fa-searchengin"></i>
+                </div>
+            </div>
+            <div className="categories-container">
+                {categoryList.map((category, index) => <CategoryChip key={index} props={category} />)}
+            </div>
+            <div className='video-layout'>
+                {videoList.map(video => {
+                    if (videoCategory === video.category) {
```

Comment:
Suggestion: Since both if and if else returns the same thing you can manage with one if
`if(condition1 || condition2)`